rubix-spi/src/main/java/com/qubole/rubix/spi/fop/ObjectPoolPartition.java

```diff
@@ -35,10 +37,11 @@
   private final PoolConfig config;
   private final BlockingQueue<Poolable<T>> objectQueue;
   private final ObjectFactory<T> objectFactory;
-  private int totalCount;
   private final String host;
   private final int socketTimeout;
   private final int connectTimeout;
+  private final Semaphore takeSemaphore;
+  private final AtomicInteger totalCount;
```

Comment:
is it possible to remove `totalCount` and rather use `takeSemaphore#permits`

Reply:
yeah, will change in next iteration.

Reply:
I checked it is not possible. We initialize the pool with min size without acquiring semaphore.